File: dottydoc/src/dotty/tools/dottydoc/api/java/Dottydoc.java
package dotty.tools.dottydoc.api.java;

import dotty.tools.dottydoc.DocDriver;
import dotty.tools.dottydoc.model.Package;
import dotty.tools.dottydoc.util.OutputWriter;
import java.util.Map;
import java.util.List;
import java.net.URL;

/** FIXME: document me! */
public class Dottydoc extends DocDriver {
    public Map<String, Package> createIndex(String[] args) {
        return compiledDocsJava(args);
    }

    public String createJsonIndex(String[] args) {
        return indexToJson(createIndex(args));
    }

    public void buildDocs(
        String outputDir,
        URL template,
        List<URL> resources,
        Map<String, Package> index
    ) {
        new OutputWriter().writeJava(index, outputDir, template, resources);
    }

    public void writeJson(Map<String, Package> index, String outputDir) {
        new OutputWriter().writeJsonJava(index, outputDir);
    }
}

File: dottydoc/src/dotty/tools/dottydoc/api/scala/Dottydoc.scala
package dotty.tools.dottydoc.api.scala

import dotty.tools.dottydoc.DocDriver
import dotty.tools.dottydoc.model.Package
import dotty.tools.dottydoc.util.OutputWriter

import scala.collection.Map
import java.net.URL

/** FIXME: document this class plz */
trait Dottydoc extends DocDriver {
  def createIndex(args: Array[String]): Map[String, Package] =
    compiledDocs(args)

  def buildDocs(outDir: String, template: URL, resources: List[URL], index: Map[String, Package]) =
    new OutputWriter().write(index, outDir, template, resources)

  def writeJson(index: Map[String, Package], outputDir: String) =
    new OutputWriter().writeJson(index, outputDir)
}
